engine/api/src/main/scala/pl/touk/nussknacker/engine/api/namespaces/ObjectNaming.scala

```diff
@@ -4,6 +4,7 @@ import com.typesafe.config.Config
 
 trait ObjectNaming extends Serializable {
   def prepareName(originalName: String, config: Config, namingContext: NamingContext): String
+  def getNamespace(config: Config, namingContext: NamingContext): Option[String] = None
```

Comment:
I think by adding this name we make some assumptions about prepareName method? Probably we want to specify them somehow? I mean, we assume that originalName + namespace will determine return value of prepareName?

Reply:
Hm yes, I made the `prepareName` function depend on `getNamespace` and `getSeparator` functions, do you think it's better this way? If somebody doesn't like the namespace + separator + originalName convention, they still can overwrite the `prepareName` function.